go/vt/vttablet/tabletmanager/vdiff/table_plan.go

```diff
@@ -183,6 +194,13 @@ func (tp *tablePlan) findPKs(targetSelect *sqlparser.Select) error {
 			}
 			if strings.EqualFold(pk, colname) {
 				tp.compareCols[i].isPK = true
+				// We need to set the correct collation, if the column has one, to ensure
+				// proper ordering of the results when we later do the merge sort.
+				collationName, err := getColumnCollation(dbClient, tp.dbName, tp.table.Name, colname)
```

Comment:
I thought about doing that too. I don't think it will make a significant difference either way, but there's no reason NOT to limit queries against the DB. I'll move to a single query and a map. Thanks!